ion-hash/src/representation.rs

```diff
@@ -120,9 +137,99 @@ fn write_repr_float<D: Update>(value: Option<f64>, hasher: &mut EscapingDigest<'
     }
 }
 
-fn write_repr_string<D: Update>(value: Option<&str>, hasher: &mut EscapingDigest<'_, D>) {
+fn write_repr_symbol<D, S>(value: Option<&S>, hasher: &mut EscapingDigest<'_, D>)
+where
+    D: Update,
+    S: SymbolToken + ?Sized,
+{
     match value {
-        Some(s) => hasher.update(s.as_bytes()),
+        Some(s) => {
+            match s.text() {
+                Some(s) => write_repr_string(Some(s), hasher),
+                None => {
+                    // FIXME: Throw an error if this is not $0
+                    {}
```

Comment:
Since this handles neither `$0` nor non-`$0`, this might be a good place to add a `todo!()` so any tests that exercise this code path will fail noisily.
```suggestion
                    todo!("hash SymbolToken without text")
```